python/cuda_cccl/cuda/compute/_caching.py

```diff
@@ -77,7 +77,11 @@ def __eq__(self, other):
         return self._identity == other._identity
 
     def __hash__(self):
-        return hash(self._identity)
+        try:
+            return hash(self._identity)
+        except TypeError:
+            # if we can't hash the identity, use its id
+            return id(self._identity)
```

Comment:
I recommend hashing the resulting `id` value.
```suggestion
            return hash(id(self._identity))
```